DS-4408 Add vignette for dAIC calc in Linear Regression

jrwishart opened the pull request.

jrwishart committed:
DS-4408 Add vignette for dAIC calc in Linear Reg

mwmclean commented on vignettes/dAIC-Linear-Regression.Rmd:
```diff
@@ -0,0 +1,158 @@
+    \mathcal{H}(\boldsymbol{\theta}) &= \begin{pmatrix} \frac{\partial^2 \ell}{\partial \boldsymbol{\beta} \partial \boldsymbol{\beta}^T} & \frac{\partial^2 \ell}{\partial \boldsymbol{\beta} \partial \sigma^2} \\ \frac{\partial^2 \ell}{\partial \sigma^2 \partial \boldsymbol{\beta}^T} & \frac{\partial^2 \ell}{\partial \sigma^2 \partial \sigma^2} \end{pmatrix}\\
+        \frac{\partial^2\ell}{\partial\boldsymbol{\beta}\partial\boldsymbol{\beta}^T} &= \frac{\partial}{\partial\boldsymbol{\beta}^T} \boldsymbol{X}^T (\boldsymbol{Y} - \boldsymbol{X}\boldsymbol{\beta}) / \sigma^2 = - \frac{\boldsymbol{X}^T\boldsymbol{X}}{\sigma^2}\\
+        \frac{\partial^2\ell}{\partial\boldsymbol{\beta}\partial\sigma^2} &= \frac{\partial}{\partial\sigma^2} \boldsymbol{X}^T \boldsymbol{\varepsilon}^T / \sigma^2 = - \frac{\boldsymbol{X}^T\boldsymbol{\varepsilon}}{(\sigma^2)^2} \Rightarrow \frac{\partial^2\ell}{\partial\sigma^2\partial\boldsymbol{\beta}^T} = - \frac{\boldsymbol{\varepsilon}^T\boldsymbol{X}}{(\sigma^2)^2}\\
+        \frac{\partial^2\ell}{\partial\sigma^2\partial\sigma^2} &= -\frac{1}{2\sigma^2} \frac{\partial}{\partial\sigma^2} (n - \boldsymbol{\varepsilon}^T \boldsymbol{\varepsilon} / \sigma^2) = \frac{n}{2(\sigma^2)^2} - \frac{\boldsymbol{\varepsilon}^T\boldsymbol{\varepsilon}}{(\sigma^2)^3}
```
Where does the middle equality come from? Derivative of a constant w.r.t. to sigma^2 isn't zero?

jrwishart replied:
Had that factor outside the partial ~derivate~ derivative by mistake. Fixed in [42482f8](https://github.com/Displayr/flipRegression/pull/62/commits/42482f83432b6ed8e656129f5fd5d5c50c0605ce)

jrwishart committed:
DS-4408 Fix typo in SRS Hessian proof

jrwishart committed:
DS-4408 Add hat on reciporical estimator for V_sigma^2

mwmclean commented on vignettes/dAIC-Linear-Regression.Rmd:
```diff
@@ -0,0 +1,158 @@
+## Weighted likelihood
+
+The parameter space and underlying model is still the same but a simple random sample is not assumed. Instead each observed unit is given a sample weight of $w_i$ to scale up each observation to reflect the occurrence in the population. More specifically, they are chosen such that $\sum_{i = 1}^n w_i = N$, the population total. Then the likelihood is estimated using the weighted estimator,
+\[
```
Then the **log** likelihood...

jrwishart replied:
Updated in f0b2eac3fbda6a7923c4092a8dd2c7e7a08210de. Thanks.

mwmclean commented on vignettes/dAIC-Linear-Regression.Rmd:
```diff
@@ -0,0 +1,158 @@
+\]
+where $\ell_i(\boldsymbol{\theta}) = \log f_{\boldsymbol{\theta}}(y_i|\boldsymbol{x}_i)$ an observation can be considered sampled with probability $1/w_i$. For convenience define $\boldsymbol{W} = \mathrm{diag}(w_i)_{i = 1}^n$ as the diagonal matrix of weights.
+
+Leveraging the earlier results from the unweighted likelihood, the likelihood takes the form,
```
log-likelihood

jrwishart replied:
Updated in f0b2eac3fbda6a7923c4092a8dd2c7e7a08210de. Thanks.

mwmclean commented on vignettes/dAIC-Linear-Regression.Rmd:
```diff
@@ -0,0 +1,158 @@
+Leveraging the earlier results from the unweighted likelihood, the likelihood takes the form,
+
+\begin{aligned}
+    \widehat{\ell}(\boldsymbol{\theta}) &= - \frac{1}{2} \log(2 \pi) - \frac{1}{2} \log \sigma^2 -  \frac{1}{N}\frac{(\boldsymbol{Y} - \boldsymbol{X}\boldsymbol{\beta})^T\boldsymbol{W}(\boldsymbol{Y} - \boldsymbol{X}\boldsymbol{\beta})}{2\sigma^2}\\
```
What happened to the N on the first and second term? Souldn't the second term involve the weights?

jrwishart replied:
$\displaystyle \sum_{i=1}^{n} w_i = N$ so the $N$ values cancel out.

mwmclean replied:
Should mention that here in the text instead of leaving it until the implementation section at the end.

jrwishart replied:
The weights aggregating to the population total is separate to the rescaling at the end. I'll amend the statement at the end

mwmclean replied:
If the weights didn't sum to N, then they would appeaer in the first two terms in the likelihood. The cancellation occurs because of the assumption.

jrwishart committed:
DS-4408 Write log-likelihood instead of likelihood

jrwishart committed:
DS-4408 Change to PDF

jrwishart committed:
DS-4408 Notation improvements

jrwishart committed:
DS-4408 Add to-do list

jrwishart committed:
DS-4408 Add basic sampling design motivation

jrwishart committed:
DS-4408 Move hats on delta estimator

jrwishart committed:
fixup! DS-4408 Move hats on delta estimator

jrwishart committed:
DS-4408 Rewrite first section to be a pure MLE approach

jrwishart committed:
DS-4408 Dont use bold sigma

jrwishart committed:
DS-4408 Tidy up unweighted case to make dAIC motivation clearer

jrwishart committed:
DS-4408 Avoid notation clash

Redefine the sample total log likelihood using \mathcal L instead of
\ell to avoid notational confusion in the weighted case.

jrwishart committed:
DS-4408 WIP Define the complex survey context in a better way

jrwishart committed:
fixup! DS-4408 Avoid notation clash

jrwishart committed:
fixup! DS-4408 WIP Define the complex survey context in a better way

jrwishart committed:
DS-4408 Next version of proof

jrwishart committed:
DS-4408 Add notes about survey implementation

jrwishart committed:
fixup! DS-4408 Add notes about survey implementation

mwmclean commented on vignettes/dAIC-Linear-Regression.Rmd:
```diff
@@ -0,0 +1,218 @@
+\]
+where the contribution to the likelihood by unit $i$ is given by $\ell_i(\boldsymbol{\theta}) = \log f_{\boldsymbol{\theta}}(y_i |\boldsymbol{x}_i)$.
+
+Then $\boldsymbol{\theta}^*$ is estimated by $\widehat{\boldsymbol{\theta}}$ that satisfies the is the weighted sample score equations, $\widehat{\mathcal U}(\widehat{\boldsymbol{\theta}}) = 0$.
```
Can remove "the is"

jrwishart replied:
Fixed in 93fdf68

jrwishart committed:
DS-4408 Fix grammar/typos from code review

mwmclean commented on vignettes/dAIC-Linear-Regression.Rmd:
```diff
@@ -0,0 +1,218 @@
+
+This is motivated by the using the Kullback-Leibler divergence of the candidate parametric model away from the true model,
+\[
+    \mathrm{KL}(f_{\widehat{\boldsymbol{\theta}}}, g) = \mathbb{E}_g \left( \log g(y| \boldsymbol{x})\right) - \ell(\boldsymbol{\theta}).
```
Not sure why there's a `\widehat` here

jrwishart replied:
Fixed in 07740298f974df121c532a88dfa15942dddff2e0

jrwishart committed:
DS-4360 Further updates

mwmclean commented on vignettes/dAIC-Linear-Regression.Rmd:
```diff
@@ -0,0 +1,218 @@
+    \mathrm{KL}(f_{\boldsymbol{\theta}}, g) = \mathbb{E}_g \left( \log g(y| \boldsymbol{x})\right) - \ell(\boldsymbol{\theta}).
+\]
+
+The target parameter is $\boldsymbol{\theta}^*$ is the parametric value that is the 'least false' which is the value that minimizes the Kullback-Leibler above (or equivalently, maximizer of $\ell(\boldsymbol{\theta}))$. In terms of the classical likelihood equations, this occurs at the new score equation,
```
Kullback-Leibler divergence

jrwishart replied:
Fixed in 288003f

mwmclean commented on vignettes/dAIC-Linear-Regression.Rmd:
```diff
@@ -0,0 +1,218 @@
+
+Then $\boldsymbol{\theta}^*$ is estimated by $\widehat{\boldsymbol{\theta}}$ that satisfies the weighted sample score equations, $\widehat{\mathcal U}(\widehat{\boldsymbol{\theta}}) = 0$.
+
+Before computing the required terms in this analysis, the link between $\widehat{\boldsymbol{\theta}}$ and the AIC strategy of model assessment is covered. The model that matches the population likelihood the best will minimize the Kullback-Leibler above. The AIC strategy is to estimate this difference by estimating $Q_n = \mathbb{E}_g( \ell(\widehat{\boldsymbol{\theta}}))$ and selecting the mode ...
```
Kullback-Leibler divergence

jrwishart replied:
Fixed in 288003f

jrwishart committed:
DS-4408 Add divergence to KL statements

mwmclean commented on vignettes/dAIC-Linear-Regression.Rmd:
```diff
@@ -0,0 +1,218 @@
+
+Then $\boldsymbol{\theta}^*$ is estimated by $\widehat{\boldsymbol{\theta}}$ that satisfies the weighted sample score equations, $\widehat{\mathcal U}(\widehat{\boldsymbol{\theta}}) = 0$.
+
+Before computing the required terms in this analysis, the link between $\widehat{\boldsymbol{\theta}}$ and the AIC strategy of model assessment is covered. The model that matches the population likelihood the best will minimize the Kullback-Leibler divergence above. The AIC strategy is to estimate this difference by estimating $Q_n = \mathbb{E}_g( \ell(\widehat{\boldsymbol{\theta}}))$ and selecti ...
```
... of the naive estimator, $\widehat{\ell}(\widehat{\boldsymbol{\theta}})$, with

jrwishart replied:
Can you give more detail here? I can't deduce which improvement you are suggesting.

mwmclean replied:
Just want to be clear that \widehat{\ell}(\widehat{\boldsymbol{\theta}}}) *is* the naive estimator and we're not talking about a naive estimator of the naive estimator.

jrwishart replied:
Fixed in 857a30a

mwmclean commented on vignettes/dAIC-Linear-Regression.Rmd:
```diff
@@ -0,0 +1,218 @@
+\]
+where the contribution to the log-likelihood by unit $i$ is given by $\ell_i(\boldsymbol{\theta}) = \log f_{\boldsymbol{\theta}}(y_i |\boldsymbol{x}_i)$.
+
+Then $\boldsymbol{\theta}^*$ is estimated by $\widehat{\boldsymbol{\theta}}$ that satisfies the weighted sample score equations, $\widehat{\mathcal U}(\widehat{\boldsymbol{\theta}}) = 0$.
```
pseudo-score equation

jrwishart replied:
Fixed in b12d6f2db43d1e3e3aba5f9ade70de2e606af826

mwmclean commented on vignettes/dAIC-Linear-Regression.Rmd:
```diff
@@ -0,0 +1,218 @@
+\]
+where the contribution to the log-likelihood by unit $i$ is given by $\ell_i(\boldsymbol{\theta}) = \log f_{\boldsymbol{\theta}}(y_i |\boldsymbol{x}_i)$.
+
+Then $\boldsymbol{\theta}^*$ is estimated by $\widehat{\boldsymbol{\theta}}$ that satisfies the weighted sample score equations, $\widehat{\mathcal U}(\widehat{\boldsymbol{\theta}}) = 0$.
```
*consistently* estimated by ... 
Then can move citation to Fuller here

jrwishart replied:
Fixed in b12d6f2db43d1e3e3aba5f9ade70de2e606af826

mwmclean commented on vignettes/dAIC-Linear-Regression.Rmd:
```diff
@@ -0,0 +1,218 @@
+In particular, their proof utilizes the regularity conditions and results of Theorem 1.3.9 [@Fuller-2009] under the typical framework in survey estimation where it is supposed that there are a sequence of finite populations indexed by $\nu$ and a sequence of samples of size $n_\nu$ drawn from the $N_\nu$ units in the $\nu^{\mathrm th}$ population with a well-defined probability scheme. Then asymp ...
+
+1. $\widehat{\boldsymbol{\theta}}_\nu$ is asymptotically normal with $\sqrt{n} (\widehat{\boldsymbol{\theta}}_{\nu}- \boldsymbol{\theta}^*) \stackrel{d}{\longrightarrow} \mathcal N(\boldsymbol{0}, \boldsymbol{V}(\boldsymbol{\theta}^*))$.
+2. If a sequence $\left\{\boldsymbol{\theta}_\nu \right\}$ are consistent estimators of $\boldsymbol{\theta}^*$, that is,$\boldsymbol{\theta}_\nu \stackrel{p}{\longrightarrow} \boldsymbol{\theta}^*$, then $\widehat{\mathcal J}_\nu(\boldsymbol{\theta}_\nu) \stackrel{p}{\longrightarrow} \mathcal I(\boldsymbol{\theta})$.
```
Missing space after "that is,"

jrwishart replied:
Fixed in a9fee89eb978fdb5dcb11864b087d89c89b89c42

mwmclean commented on vignettes/dAIC-Linear-Regression.Rmd:
```diff
@@ -0,0 +1,218 @@
+
+### Implementation
+
+This result is implemented using the [@survey] package to compute the model but $d$AIC computed manually.
```
using the \texttt{survey) package [@survey]

jrwishart replied:
Fixed in a9fee89eb978fdb5dcb11864b087d89c89b89c42

jrwishart committed:
DS-4408 Add space and tidy survey referencing

jrwishart committed:
DS-4408 Rephrase definition of weight aggregation total

jrwishart committed:
DS-4408 Improve def of pseudo-MLE equations

jrwishart committed:
DS-4408 Amend statement on rescaling of weights

jrwishart committed:
DS-4408 Fix grammar in referring to estimator

jrwishart committed:
DS-4408 Improve order of statements at end of weighted likelihood section

jrwishart committed:
DS-4408 Fix type in eps definition

jrwishart committed:
DS-4408 Don't use sampling notation in unweighted section

jrwishart committed:
DS-4408 and DS-4534 Update documentation

jrwishart committed:
DS-4408 Increment version and add vignette builder

jrwishart committed:
DS-4408 Fix signs in Jhat and JN
